src/middleware/redux.ts

```diff
@@ -3,27 +3,44 @@ import type { NamedSet } from './devtools'
 
 type Write<T, U> = Omit<T, keyof U> & U
 
-type Action = { type: unknown }
+type LitObjAction = {
+  type: unknown
+} & Partial<{ [key: string]: unknown }>
+type TupleAction = [unknown, unknown?]
+type Action = LitObjAction | TupleAction
+type Tupleize<T> = T extends LitObjAction
+  ? [T]
+  : T extends TupleAction
+  ? T
+  : never
+
+type StoreRedux<AIn, AOut> = {
+  dispatch: (...a: Tupleize<AIn>) => AOut
```

Comment:
I wonder it can be something like this:
```suggestion
type StoreRedux<A, Rest> = {
  dispatch: (a: A, ...rest: Rest) => A
```